XenAdmin/Controls/SrPickerItem.cs

```diff
@@ -91,35 +88,19 @@ public SrPickerMigrateItem(SR sr, Host aff, long diskSize, VDI[] vdis) : base(sr
         {
         }
 
-        private bool CanMigrate(VDI vdi)
-        {
-            if (TheSR == null || vdi == null)
-                return false;
-
-            bool toLocal = TheSR.IsLocalSR();
-            if (toLocal && !HomeHostCanSeeTargetSr(vdi))
-                return false;
-
-            bool fromLocal = vdi.Connection.Resolve(vdi.SR).IsLocalSR();
-            if (fromLocal && toLocal)
-                return false;
-
-            return true;
-        }
-
         /// <summary>
-        /// If the VM has a home host, we can move VDIs to a local SR only
-        /// if the latter belongs to the VM's home host
+        /// We can move VDIs to a local SR only if the VDI is attached to VMs
+        /// that have a home server that can see the SR
         /// </summary>
-        private bool HomeHostCanSeeTargetSr(VDI vdi)
+        private static bool HomeHostCanSeeTargetSr(VDI vdi, SR targetSr)

```

Comment:
Why was this method made static? Currently it is only used inside this class and with TheSR passed to the targetSr parameter.

Reply:
Long ago that one used to be instance method `SrIsLocalToTheHostOnWhichTheVmIsRunning(SR sr, VDI vdi)` and I made it instance method `HomeHostCanSeeTargetSr(VDI vdi)` for this very reason. But now I've changed my mind and I think the name is more descriptive of what it does if it is `HomeHostCanSeeTargetSr(VDI vdi, SR targetSr)`. It doesn't need to be static, I only made it static to avoid people using TheSR in it since we pass targetSr.

Reply:
OK